hydra-cluster/src/Hydra/Cluster/Scenarios.hs

```diff
@@ -179,14 +189,31 @@ refuelIfNeeded ::
   Actor ->
   Lovelace ->
   IO ()
-refuelIfNeeded tracer node actor amount = do
-  (actorVk, _) <- keysFor actor
-  (fuelUTxO, otherUTxO) <- queryMarkedUTxO node actorVk
-  traceWith tracer $ StartingFunds{actor = actorName actor, fuelUTxO, otherUTxO}
+refuelIfNeeded tracer node receiver amount = do
+  (receivingVk, _) <- keysFor receiver
+  (senderVk, senderSk) <- keysFor Faucet
+  (fuelUTxO, otherUTxO) <- queryMarkedUTxO node receivingVk
+  traceWith tracer $ StartingFunds{actor = actorName receiver, fuelUTxO, otherUTxO}
   let fuelBalance = selectLovelace $ balance @Tx fuelUTxO
   when (fuelBalance < amount) $ do
-    utxo <- seedFromFaucet node actorVk amount Fuel (contramap FromFaucet tracer)
-    traceWith tracer $ RefueledFunds{actor = actorName actor, refuelingAmount = amount, fuelUTxO = utxo}
+    utxo <- sendFundsTo node (senderVk, senderSk) receivingVk amount Fuel (contramap FromFaucet tracer) True
```

Comment:
Should: use the better named and higher-level API (using the actor type) `seedFromFaucet` here.

It's not clear to me why the lower-level `sendFundsTo` was used here. The whole `refuelIfNeeded` should stay as it was.